src/datastore/database.cpp

```diff
@@ -0,0 +1,223 @@
+        // Set's deleted bool
+        tempTeam.hidden = std::stoi(team[2]);
+        // Initializing a new Stadium class
+        Stadium tempStadium(std::stoi(team[3]), team[4], team[6], std::stoi(team[5]),std::stoi(team[9]),
+                            std::stoi(team[10]), tempRoof, tempSurface, tempTypology);
+        // Souvenir index starts at 14 for each team, reseting to 14 here
+        // Used to traverser through souvenirs
+        unsigned int j = 14;
+
+        // Index 13 holds the number of souvenirs a stadium has
+        souvenirCount = std::stoi(team[13]);
+        for(int i = 0; i < souvenirCount; ++i)
+        {
+            tempHidden = std::stoi(team[j++]); // Is hidden? - bool
+            tempName = team[j++];              // Souvenir Name
+            tempPrice = std::stod(team[j++]);  // Souvenir Price
+
+            tempStadium.addSouvenir(tempName, tempPrice); // Creating an instance of a souvenir class
+            tempStadium.findSouvenir(i).hidden = tempHidden;
+        }
+
+        // Insterting team and stadium in to their appropriate maps for database
+        teams[tempTeam.m_id] = tempTeam;
+        stadiums[tempStadium.m_id] = tempStadium;
+    }
+}
+
+/**
+ * This method returns all the teams
+ *
+ * @return teams is returned
+ */
+const nstd::map<int,Team> Database::getTeams()
+{
+    return teams;
+}
+
+/**
+ * This method returns all the Stadiums
+ *
+ * @return stadiums is returned
+ */
+const nstd::map<int,Stadium> Database::getStadiums()
+{
+    return stadiums;
+}
+
+/**
+ * Finds team by it's id.
+ *
+ * @param id an int id used to find a Team
+ * @return A const reference to a Team object
+ */
+Team& Database::findTeamById(int id)
+{
+    return teams[id];
+}
+
+/**
+ * Finds Stadium by it's id.
+ *
+ * @param id an int id used to find a Stadium
+ * @return A const reference to a Stadium object
+ */
+Stadium& Database::findStadiumById(int id)
+{
+    return stadiums[id];
+}
+
+
+/**
+ * Returns a vector with all the Teams
+ *
+ * @return A vector with all the teams
+ */
+std::vector<Team> Database::getTeamsVector()
+{
+    std::vector<Team> vec;
+
+    for(auto team : teams)
+        vec.push_back(team);
+
+    return vec;
+}
+
+/**
+ * Returns a vector with all the Stadiums
+ *
+ * @return A vector with all the stadiums
+ */
+std::vector<Stadium> Database::getStadiumsVector()
+{
+    std::vector<Stadium> vec;
+
+    for(auto team : teams)
+        vec.push_back(stadiums[team.getStadiumId()]);
+
+    return vec;
+}
+
+/**
+ * Uses the data within this class to write out the data
+ * to a csv file.
+ *
+ * @param path Where the program will write the csv file.
+ */
+void Database::saveToFile(const std::string& path)
+{
+    /* Get all team and stadium data, create a temp vector that will be used to store souvenir data */
+    std::vector<Team> teamVect(Database::getTeamsVector());
+    std::vector<Stadium> stadiumVect(Database::getStadiumsVector());
+    std::vector<Souvenir> tempSouvenirVect;
```

Comment:
Can move this down to where it's used?

Reply:
k

Reply:
Only team souvenir, not need to keep getting teams and stadiums every loop, just needs to do it once